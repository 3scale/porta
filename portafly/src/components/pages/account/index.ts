export { default as AccountPage } from './AccountPage'
export * from './ApplicationsTabContent'
export * from './ApplicationsTabDataListTable'
export * from './OverviewTabContent'
export * from './ProductLink'
