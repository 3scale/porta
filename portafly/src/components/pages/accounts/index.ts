export { default as AccountsIndexPage } from './AccountsIndexPage'
export * from './AccountsTable'
export * from './AccountsDataListTable'
export * from './ActionButtonImpersonate'
export * from './CreateAccountButton'
export * from './ExportAccountsButton'
export * from './AccountOverviewLink'
export * from './utils'
