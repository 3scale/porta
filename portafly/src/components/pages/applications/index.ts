export { default as ApplicationsIndexPage } from './ApplicationsIndexPage'
export * from './ApplicationsDataListTable'
export * from './ApplicationsTable'
export * from './CreateApplicationButton'
export * from './ApplicationPageLink'
