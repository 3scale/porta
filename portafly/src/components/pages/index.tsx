export * from './accounts'
export { default as Overview } from './Overview'
export * from './applications'
export { default as Login } from './Login'
export * from './plans'
export * from './product'
