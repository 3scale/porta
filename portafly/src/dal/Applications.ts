import { useFetch } from 'react-async'
import { IApplication } from 'types'

/**
 * Get all Applications asynchronously
 * @returns An object containing: {Â applications, error, isPending }
 */
function useGetApplications () {
  const { data, error, isPending } = useFetch<[IApplication]>(
    '/applications',
    { headers: { Accept: 'application/json' } }
  )

  return { applications: data, error, isPending }
}

export { useGetApplications }
