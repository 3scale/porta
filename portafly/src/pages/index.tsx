export { default as Overview } from './Overview'
export { default as Applications } from './Applications'
export { default as DeveloperAccounts } from './DeveloperAccounts'
export { default as Login } from './Login'
