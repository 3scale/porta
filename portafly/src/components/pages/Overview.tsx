/* eslint-disable no-console */
import React from 'react'
import { useTranslation } from 'i18n/useTranslation'
import {
  useA11yRouteChange,
  useDocumentTitle,
  useAlertsContext,
  Toolbar,
  DataListProvider,
  SearchWidget,
  PaginationWidget,
  useDataListPagination,
  useDataListTable,
  BulkSelectorWidget,
  BulkActionsWidget,
  ChangeStateModal,
  SendEmailModal,
  useDataListBulkActions,
  filterRows,
  useDataListFilters
} from 'components'
import {
  PageSection,
  TextContent,
  Title,
  Text,
  Card,
  CardBody,
  Button,
  ToolbarItem,
  ToolbarContent
} from '@patternfly/react-core'
import { BellIcon } from '@patternfly/react-icons'
import {
  TableHeader, TableBody, Table, OnSort, sortable
} from '@patternfly/react-table'
import { CategoryOption } from 'types'
import { StateLabel, useDataListData } from 'components/shared'

const categories = [
  {
    name: 'admin',
    humanName: 'Admin'
  },
  {
    name: 'group',
    humanName: 'Organization / Group'
  },
  {
    name: 'state',
    humanName: 'State',
    options: [
      {
        name: 'approved',
        humanName: 'Active'
      },
      {
        name: 'pending',
        humanName: 'Pending'
      }
    ]
  }
]

const apiData = new Array(1000).fill(0).map((_, i) => ({
  id: i,
  adminName: `Admin ${Math.random() * (i + 1)}`,
  createdAt: i,
  updatedAt: i,
  state: (categories[2] as any).options[i % 2].name,
  orgName: `Group ${Math.random() * (i + 1)}`
}))

const tableData = {
  columns: categories.map((c) => ({
    categoryName: c.name,
    title: c.humanName,
    transforms: [sortable]
  })),
  rows: apiData.map((account) => ({
    id: account.id,
    cells: [
      account.adminName,
      account.orgName,
      {
        stringValue: account.state,
        title: <StateLabel state={account.state} />
      }
    ],
    selected: false
  }))
}

const DataListTable = () => {
  const { startIdx, endIdx } = useDataListPagination()
  const {
    columns,
    rows,
    sortBy,
    setSortBy,
    selectOne,
    selectAll
  } = useDataListTable()
  const { filters } = useDataListFilters()
  const { data } = useDataListData()
  const { modal } = useDataListBulkActions()
  console.log('FetchedData: ', data)
  const onSort: OnSort = (event, index, direction) => {
    setSortBy(index, direction, true)
  }

  const filteredRows = filterRows(rows, filters, columns)
  const pageRows = filteredRows.slice(startIdx, endIdx)
  const states = categories[2].options as CategoryOption[]

  return (
    <>
      <Table
        aria-label="data-list-table"
        cells={columns}
        rows={pageRows}
        onSelect={(_ev, selected, _rowIndex, rowData) => (_rowIndex === -1
          ? selectAll(selected)
          : selectOne(rowData.id, selected)
        )}
        canSelectAll
        sortBy={sortBy}
        onSort={onSort}
      >
        <TableHeader />
        <TableBody />
      </Table>
      {modal === 'sendEmail' && <SendEmailModal items={['1', '2', '3', '4', '5', '6']} />}
      {modal === 'changeState' && <ChangeStateModal items={['1', '2', '3', '4', '5', '6']} states={states} />}
    </>
  )
}

const Overview: React.FunctionComponent = () => {
  const { t } = useTranslation('overview')

  useA11yRouteChange()
  useDocumentTitle(t('page_title'))
  const { addAlert } = useAlertsContext()

  const actions = {
    sendEmail: t('shared:bulk_actions.send_email'),
    changeState: t('shared:bulk_actions.change_state')
  }

  const dataListInitialState = {
    data: apiData,
    filters: {},
    table: tableData
  }

  return (
    <>
      <PageSection variant="light">
        <TextContent>
          <Title headingLevel="h2" size="3xl">{t('body_title')}</Title>
          <Text>
            {t('subtitle')}
          </Text>
        </TextContent>
      </PageSection>
      <PageSection>
        <Card>
          <CardBody>
            <TextContent>
              <p>{t('shared:format.uppercase', { text: 'Ohai' })}</p>
              {/* This is just for testing, will be removed */}
              <Button
                icon={<BellIcon />}
                onClick={() => {
                  const id = Date.now().toString()
                  addAlert({ id, variant: 'info', title: `Test Alert ${id}` })
                }}
              >
                Add an Alert
              </Button>
            </TextContent>
          </CardBody>
        </Card>
        <Card>
          <CardBody>
            <DataListProvider initialState={dataListInitialState}>
              <Toolbar>
                <ToolbarContent>
                  <ToolbarItem>
                    <BulkSelectorWidget />
                  </ToolbarItem>
                  <ToolbarItem>
                    <BulkActionsWidget actions={actions} />
                  </ToolbarItem>
                  <ToolbarItem>
                    <SearchWidget categories={categories} />
                  </ToolbarItem>
                  <ToolbarItem>
                    <PaginationWidget itemCount={tableData.rows.length} />
                  </ToolbarItem>
                </ToolbarContent>
              </Toolbar>
              <DataListTable />
            </DataListProvider>
          </CardBody>
        </Card>
      </PageSection>
    </>
  )
}

// Default export needed for React.lazy
// eslint-disable-next-line import/no-default-export
export default Overview
