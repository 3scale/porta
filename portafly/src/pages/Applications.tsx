import React from 'react'
import { useDocumentTitle, Loading, ApplicationsTable } from 'components'
import {
  Alert,
  PageSection,
  PageSectionVariants,
  TextContent,
  Text,
  TextVariants,
} from '@patternfly/react-core'
import { useGetApplications } from 'dal/Applications'

const Applications: React.FunctionComponent = () => {
  useDocumentTitle('Applications')

  const { applications, error, isPending } = useGetApplications()

  return (
    <>
      <PageSection variant={PageSectionVariants.light}>
        <TextContent>
          <Text component="h1">Applications</Text>
        </TextContent>
        <TextContent>
          <Text component={TextVariants.p}>
            This is the applications screen.
          </Text>
        </TextContent>
      </PageSection>

      <PageSection>
        {isPending && <Loading />}
        {error && <Alert variant="danger" title={error.message} />}
        {applications && <ApplicationsTable applications={applications} />}
      </PageSection>
    </>
  )
}

export default Applications
