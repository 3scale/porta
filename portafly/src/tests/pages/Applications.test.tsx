import * as React from 'react'

import { render } from 'tests/setup'
import { Applications } from 'pages'
import { useGetApplications } from 'dal/Applications'
import { getByLabelText } from '@testing-library/react'

jest.mock('dal/Applications')

export const applications = [{
  name: 'Application Name',
  state: 'Some status',
  account: 'Developer',
  plan: { name: 'Plan Name' },
  created_at: 1234567890
}]

describe('when the request is pending', () => {
  (useGetApplications as jest.Mock).mockReturnValueOnce({ isPending: true })

  it('should render spinner', () => {
    const wrapper = render(<Applications />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('when backend returns an error', () => {
  (useGetApplications as jest.Mock).mockReturnValueOnce({ error: { message: 'ERROR'} })

  it('should render an alert', () => {
    const wrapper = render(<Applications />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('when backend returns a list of applications', () => {
  (useGetApplications as jest.Mock).mockReturnValueOnce({ applications })

  it('should render a table with applications', () => {
    const wrapper = render(<Applications />)
    expect(wrapper).toMatchSnapshot()
  })
})
