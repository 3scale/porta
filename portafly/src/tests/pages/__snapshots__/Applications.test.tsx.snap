// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when backend returns a list of applications should render a table with applications 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Applications
          </h1>
        </div>
        <div
          class="pf-c-content"
        >
          <p
            class=""
            data-pf-content="true"
          >
            This is the applications screen.
          </p>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <table
          aria-label="Applications list"
          class="pf-c-table pf-m-grid-md"
          role="grid"
        >
          <thead
            class=""
          >
            <tr>
              <th
                class=""
                data-key="0"
                data-label="Name"
                scope="col"
              >
                Name
              </th>
              <th
                class=""
                data-key="1"
                data-label="State"
                scope="col"
              >
                State
              </th>
              <th
                class=""
                data-key="2"
                data-label="Account"
                scope="col"
              >
                Account
              </th>
              <th
                class=""
                data-key="3"
                data-label="Plan"
                scope="col"
              >
                Plan
              </th>
              <th
                class=""
                data-key="4"
                data-label="Created at"
                scope="col"
              >
                Created at
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr
              class=""
            >
              <td
                class=""
                data-key="0"
                data-label="Name"
              >
                Application Name
              </td>
              <td
                class=""
                data-key="1"
                data-label="State"
              >
                Some status
              </td>
              <td
                class=""
                data-key="2"
                data-label="Account"
              >
                Developer
              </td>
              <td
                class=""
                data-key="3"
                data-label="Plan"
              >
                Plan Name
              </td>
              <td
                class=""
                data-key="4"
                data-label="Created at"
              >
                1234567890
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          Applications
        </h1>
      </div>
      <div
        class="pf-c-content"
      >
        <p
          class=""
          data-pf-content="true"
        >
          This is the applications screen.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <table
        aria-label="Applications list"
        class="pf-c-table pf-m-grid-md"
        role="grid"
      >
        <thead
          class=""
        >
          <tr>
            <th
              class=""
              data-key="0"
              data-label="Name"
              scope="col"
            >
              Name
            </th>
            <th
              class=""
              data-key="1"
              data-label="State"
              scope="col"
            >
              State
            </th>
            <th
              class=""
              data-key="2"
              data-label="Account"
              scope="col"
            >
              Account
            </th>
            <th
              class=""
              data-key="3"
              data-label="Plan"
              scope="col"
            >
              Plan
            </th>
            <th
              class=""
              data-key="4"
              data-label="Created at"
              scope="col"
            >
              Created at
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr
            class=""
          >
            <td
              class=""
              data-key="0"
              data-label="Name"
            >
              Application Name
            </td>
            <td
              class=""
              data-key="1"
              data-label="State"
            >
              Some status
            </td>
            <td
              class=""
              data-key="2"
              data-label="Account"
            >
              Developer
            </td>
            <td
              class=""
              data-key="3"
              data-label="Plan"
            >
              Plan Name
            </td>
            <td
              class=""
              data-key="4"
              data-label="Created at"
            >
              1234567890
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`when backend returns an error should render an alert 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Applications
          </h1>
        </div>
        <div
          class="pf-c-content"
        >
          <p
            class=""
            data-pf-content="true"
          >
            This is the applications screen.
          </p>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <div
          aria-label="Danger Alert"
          class="pf-c-alert pf-m-danger"
        >
          <div
            class="pf-c-alert__icon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                transform=""
              />
            </svg>
          </div>
          <h4
            class="pf-c-alert__title"
          >
            <span
              class="pf-u-screen-reader"
            >
              Danger alert:
            </span>
            ERROR
          </h4>
          
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          Applications
        </h1>
      </div>
      <div
        class="pf-c-content"
      >
        <p
          class=""
          data-pf-content="true"
        >
          This is the applications screen.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-danger"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              transform=""
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          ERROR
        </h4>
        
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`when the request is pending should render spinner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-pf-content="true"
          >
            Applications
          </h1>
        </div>
        <div
          class="pf-c-content"
        >
          <p
            class=""
            data-pf-content="true"
          >
            This is the applications screen.
          </p>
        </div>
      </section>
      <section
        class="pf-c-page__main-section"
      >
        <section
          aria-label="Loading Content Container"
          class="pf-c-page__main-section"
        >
          <div
            class="pf-c-empty-state"
          >
            <div
              class="pf-c-empty-state__icon"
            >
              <span
                aria-valuetext="Loading..."
                class="pf-c-spinner pf-m-xl"
                role="progressbar"
              >
                <span
                  class="pf-c-spinner__clipper"
                />
                <span
                  class="pf-c-spinner__lead-ball"
                />
                <span
                  class="pf-c-spinner__tail-ball"
                />
              </span>
            </div>
            <h1
              class="pf-c-title pf-m-lg"
            >
              Loading
            </h1>
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <div
        class="pf-c-content"
      >
        <h1
          class=""
          data-pf-content="true"
        >
          Applications
        </h1>
      </div>
      <div
        class="pf-c-content"
      >
        <p
          class=""
          data-pf-content="true"
        >
          This is the applications screen.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section"
    >
      <section
        aria-label="Loading Content Container"
        class="pf-c-page__main-section"
      >
        <div
          class="pf-c-empty-state"
        >
          <div
            class="pf-c-empty-state__icon"
          >
            <span
              aria-valuetext="Loading..."
              class="pf-c-spinner pf-m-xl"
              role="progressbar"
            >
              <span
                class="pf-c-spinner__clipper"
              />
              <span
                class="pf-c-spinner__lead-ball"
              />
              <span
                class="pf-c-spinner__tail-ball"
              />
            </span>
          </div>
          <h1
            class="pf-c-title pf-m-lg"
          >
            Loading
          </h1>
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
